ADAL/src/ADInstanceDiscovery.m

```diff
@@ -149,9 +136,11 @@ -(void) validateAuthority: (NSString*) authority
     NSString* message = [NSString stringWithFormat:@"Attempting to validate the authority: %@; CorrelationId: %@", authority, [correlationId UUIDString]];
     AD_LOG_VERBOSE(@"Instance discovery", correlationId, message);
     
-    ADAuthenticationError* error;
+    authority = [authority lowercaseString];
```

Comment:
lowercaseString doesn't start with copy, alloc, new or mutableCopy so we don't own the reference to it. Similarly we don't own objects passed to us in parameters